.github/workflows/haskell.yml

```diff
@@ -160,14 +160,14 @@ jobs:
     - name: Archive working directory
       # pax format is needed to avoid unnecessary rebuilding, because the
       # default format uses a one-second resolution for timestamps
-      run: tar --format pax -cf /var/tmp/state.tar . && mv /var/tmp/state.tar .
+      run: tar --absolute-names --use-compress-program zstdmt --format pax -cf /var/tmp/state.tzst . && mv /var/tmp/state.tzst .
```

Comment:
What's the reason for the absolute names? It could mean that the archive won't restore correctly on a different VM that uses a different path for the build directory.

Reply:
That was from my attempt when I also tried storing cabal store, which is located in the user directory. From my experience with github actions the user name is always the same, so path does not change between VMs. However, not that I dropped storing of cabal store back in favor of cache, absolute names are no longer needed.
Removed this change too